framework/src/partitioner/MoosePartitioner.C

```diff
@@ -33,3 +33,9 @@ MoosePartitioner::MoosePartitioner(const InputParameters & params) :
 MoosePartitioner::~MoosePartitioner()
 {
 }
+
+Partitioner *
+MoosePartitioner::getPartitioner()
+{
+  return static_cast<Partitioner *>(this);
+}
```

Comment:
What's this needed for?  Generally you don't need a special function to cast to the base class...